modules/CuckooFilter/src/main.rs

```diff
@@ -1,3 +1,57 @@
-fn main() {
-    println!("Hello, world!");
-}
+extern crate redis;
+use redis::Commands;
+use probabilistic_collections::cuckoo::CuckooFilter;
+
+fn main() {
+    let size = std::env::args().nth(1).expect("Please specify the size of the filter");
+    let action = std::env::args().nth(2).expect("Please specify an action");
+    let item = std::env::args().nth(3).expect("Please specify an item");
+    
+    println!("size of the filter {:?}", size);
+
+    let mut cuckoo_filter = CuckooFilter::<String>::new(size.parse::<usize>().unwrap());
+
+    if action == "add"{

```

Comment:
**[Desired, Maintainability]** It's not good practise to match against raw strings. This is why enums and the `match` statement exist. It would be better to define the `FromStr` trait for the Command enum and then match against the resolved value. Would be good too read up on traits and how they work. For example:
```rust
#[derive(Debug, PartialEq, Eq)]
enum Commands {
    ADD,
    DELETE,
}

#[derive(Debug)]
struct CommandParsingError {
    msg: &'static str,
}

impl fmt::Display for CommandParsingError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "Could not parse Command for string: {}", self.msg)
    }
}

impl Error for CommandParsingError {
    fn source(&self) -> Option<&(dyn Error + 'static)> {
        Some(&self.msg)
    }
}

impl FromStr for Commands {
    type Err = CommandParsingError;
    fn from_str(s: &str) -> Result<Self, CommandParsingError> {
        let upper_s: String = s.to_uppercase();
        if upper_s == "ADD" {
            return Ok(Commands::ADD);
        } else if upper_s == "DELETE" {
            return Ok(Commands::DELETE);
        }
        return Err(CommandParsingError{msg: s})
    }
}

let parsed_command: Command = Command::from_str(CuckooFilter::<String>::new(size.parse::<usize>().unwrap()))?;
```

Reply:
**[Required]** You will need to write some tests for the commands too. Currently, there's nothing here that tells me it works. We need to be able to test changes that are made to your code in future as well as ensure that it works now.

Reply:
@alooba1 Why do you put this as resolved? Couldn't see the implementation of Jack's suggested Code change.